src/lib/fy-parse.c

```diff
@@ -3171,13 +3172,15 @@ int fy_fetch_block_scalar(struct fy_parser *fyp, bool is_literal, int c)
 	int actual_lb_length, pending_lb_length;
 	struct fy_mark indicator_mark;
 	bool generated_indent;
+	char *str;
 #ifdef ATOM_SIZE_CHECK
 	size_t tlength;
 #endif
 
+    fy_utf8_format_a(c, fyue_singlequote, &str);
```

Comment:
Yeah, that's not good.

You unconditionally format the error string even when no error happens.
This is on the hot path, and will slow down things.

We would probably have to do an open coded if () followed by a call to fyp_error()